components/config/src/config/mod.rs

```diff
@@ -98,6 +98,9 @@ pub struct Config {
     pub markdown: markup::Markdown,
     /// All user params set in `[extra]` in the config
     pub extra: HashMap<String, Toml>,
+
+    /// Whether to exclude paginated pages in sitemap
+    pub exclude_paginated_pages_in_sitemap: bool,
```

Comment:
Bikeshedding: i don't know whether it should go on another nested level in the config since we do regret top level fields everytime.. Maybe a seo struct?

I believe it should also be an enum:
- none
- first only
- all

since i did see some people indexing only the first page.

Reply:
Let's take an example of my site https://bauva.com since I have tested this PR on it. The [blog list page](https://bauva.com/blog/) is paginated and it contains the link to the next page. The current sitemap includes `/blog/` and `/blog/page/1/` and `/blog/page/2/`. After the current PR, the sitemap includes `/blog/` which can still work for search engines to discover the other paginated pages because the "next" and "prev" buttons. Also for the current behaviour `/blog/` and `/blog/page/1/` are the exact same pages which IMO should be avoided.

If we add another option like first only, the sitemap would contain `/blog/` and `/blog/page/1/` which will cause the same non-canonical page problem.

Reply:
I think let's still make it a string because otherwise it will be a breaking change if we decide to have 3 behaviours. Maybe `paginated_pages_in_sitemap` as a name?

Reply:
Sure